app_sensor_module/src/sensors.c

```diff
@@ -0,0 +1,164 @@
+    for (uint8_t i = 0; i < num_sensors; i++) {
+        if (l_check_device(sensors[i]) == 0) {
+            sensor_ready[i] = true;
+        } else {
+            LOG_INF("Sensor %d not ready", i);
+            sensor_ready[i] = false;
+        }
+    }
+}
+
+static void
+populate_telemetry_struct(sensor_module_hundred_hz_telemetry_t *telemetry, const struct device *const *sensors,
+                          const enum sensor_channel *const *channels_arr) {
+    float sensor_values[3] = {0};
+    for (uint8_t i = 0; i < SENSOR_MODULE_NUM_HUNDRED_HZ_SENSORS; i++) {
+        switch (i) {
+            case SENSOR_MODULE_ADXL375:
+                l_get_sensor_data_float(sensors[i], 3, channels_arr[i], sensor_values);
+                telemetry->adxl375.accel_x = sensor_values[0];
+                telemetry->adxl375.accel_y = sensor_values[1];
+                telemetry->adxl375.accel_z = sensor_values[2];
+                break;
+            case SENSOR_MODULE_LSM6DSL:
+                l_get_sensor_data_float(sensors[i], 3, channels_arr[i], sensor_values);
+                telemetry->lsm6dsl_accel.accel_x = sensor_values[0];
+                telemetry->lsm6dsl_accel.accel_y = sensor_values[1];
+                telemetry->lsm6dsl_accel.accel_z = sensor_values[2];
+
+                l_get_sensor_data_float(sensors[i], 3, channels_arr[i], sensor_values);
+                telemetry->lsm6dsl_gyro.gyro_x = sensor_values[0];
+                telemetry->lsm6dsl_gyro.gyro_y = sensor_values[1];
+                telemetry->lsm6dsl_gyro.gyro_z = sensor_values[2];
+                break;
+            case SENSOR_MODULE_LIS3MDL:
+                l_get_sensor_data_float(sensors[i], 3, channels_arr[i], sensor_values);
+                telemetry->lis3mdl.mag_x = sensor_values[0];
+                telemetry->lis3mdl.mag_y = sensor_values[1];
+                telemetry->lis3mdl.mag_z = sensor_values[2];
+                break;
+            case SENSOR_MODULE_MS5611:
+                l_get_sensor_data_float(sensors[i], 2, channels_arr[i], sensor_values);
+                telemetry->ms5611.pressure = sensor_values[0];
+                telemetry->ms5611.temperature = sensor_values[1];
+                break;
+            case SENSOR_MODULE_BMP388:
+                l_get_sensor_data_float(sensors[i], 2, channels_arr[i], sensor_values);
+                telemetry->bmp388.pressure = sensor_values[0];
+                telemetry->bmp388.temperature = sensor_values[1];
+                break;
+            default:
+                LOG_ERR("Invalid sensor index");
+                break;
+        }
+    }
+}
+
+static void hundred_hz_sensor_reading_task(void *unused0, void *unused1, void *unused2) {
+    ARG_UNUSED(unused0);
+    ARG_UNUSED(unused1);
+    ARG_UNUSED(unused2);
+
+    // Initialize variables for receiving telemetry
+    sensor_module_hundred_hz_telemetry_t hundred_hz_telemetry;
```

Comment:
Allocating the memory for it I guess, not really initializing? Will just be garbage data instead of set to 0s if we did = {0}, but populate_telemetry_struct should take care of it.